src/lib.rs

```diff
@@ -1828,25 +1814,25 @@ impl<
     /// Get the viewing public keys.
     pub async fn viewing_pub_keys(&self) -> Vec<ViewerPubKey> {
         let KeystoreSharedState { model, .. } = &*self.read().await;
-        model.viewing_accounts.pub_keys()
+        model.viewing_accounts.iter_pub_keys().collect()
     }
 
     /// Get the freezing public keys.
     pub async fn freezing_pub_keys(&self) -> Vec<FreezerPubKey> {
```

Comment:
`impl Iterator`

Reply:
I tried changing this to `impl Iterator` but got the `cannot return value referencing temporary value` error. To get rid of the error, we need to add the `.collect()` back and then reconstruct the iterator (or is there a better way?), which I think defeats the purpose of the change. Its caller will need to do `.collect()` again, making this more redundant.

Reply:
Oh shoot, because it borrows from the `RwLockReadGuard` instead of borrowing from `self`. Yeah, this is probably unavoidable. Never mind then.